### PR TITLE
Flush Config to use `types.Object`

### PR DESCRIPTION
**Pull request by Tang8330**
Using `types.Object` instead of our custom type so that it supports unknown values.

I also added tests around this

### CODE REVIEW

File: internal/provider/tfmodels/deployment.go [hunk 1]
```diff
@@ -55,6 +69,17 @@ func (d Deployment) ToAPIBaseModel(ctx context.Context) (artieclient.BaseDeploym
 		return artieclient.BaseDeployment{}, diags
 	}
 
+	var flushConfig *DeploymentFlushConfig
+	flushConfigDiags := d.FlushConfig.As(ctx, &flushConfig, basetypes.ObjectAsOptions{
```
Comment (danafallon) [inline internal/provider/tfmodels/deployment.go:73]: I think we should put this in an if block: `if !d.FlushConfig.IsNull() && !d.FlushConfig.IsUnknown()`

Like we do here: https://github.com/artie-labs/terraform-provider-artie/blob/main/internal/provider/tfmodels/util.go#L46

Comment (Tang8330) [reply internal/provider/tfmodels/deployment.go:73]: sgtm

Comment (Tang8330) [reply internal/provider/tfmodels/deployment.go]: I added the check -- I actually think we don't need the if statement block, I had a test in there for null and unknown: https://github.com/artie-labs/terraform-provider-artie/pull/149/files#diff-10fe6c52a789e32b39c9fc0625f5586f91c590bc69056734d82713cde6c4eeb7

File: internal/provider/tfmodels/deployment.go [hunk 2]
```diff
@@ -55,6 +69,19 @@ func (d Deployment) ToAPIBaseModel(ctx context.Context) (artieclient.BaseDeploym
...
+		flushConfigDiags := d.FlushConfig.As(ctx, &flushConfig, basetypes.ObjectAsOptions{
+			UnhandledNullAsEmpty:    true,
+			UnhandledUnknownAsEmpty: true,
+		})
+
+		diags.Append(flushConfigDiags...)
+		if diags.HasError() {
+			return artieclient.BaseDeployment{}, diags
+		}
+	}
```
Comment (danafallon) [inline internal/provider/tfmodels/deployment.go:83]: Probably also need to move the `flushConfig.ToAPIModel()` into this block so it doesn't panic if `flushConfig` is nil

File: internal/provider/tfmodels/deployment.go (hunk 1 above)
Comment (danafallon) [reply internal/provider/tfmodels/deployment.go]: Hmm but do those options mean that it'll make an empty flush config struct, where each of the properties is just set to its zero value? If the flush config isn't provided then we want to make sure we don't overwrite the existing values by sending zeros for them - the payload we send to the api should have null for the flush config to avoid overwriting.

File: internal/provider/tfmodels/deployment.go (hunk 2 above)
Comment (Tang8330) [reply internal/provider/tfmodels/deployment.go:83]: I added a guardrail for [ToAPIModel](https://github.com/artie-labs/terraform-provider-artie/blob/8407678d1fdb389a7d8039af41c767e8d8ff38f9/internal/provider/tfmodels/deployment.go#L175-L180)

**Review by danafallon — APPROVED** [review]
